docs/api/guidelines.md

```diff
@@ -0,0 +1,342 @@
+| updated        | RFC 3339 timestamp | Time a resource was last updated |
+
+If you are creating a compound field name, use lower camel case for the JSON field with the first character
+lowercase, and capitalize the first letter of each subsequent word (e.g. thisIsAField). Do not use characters
+other that ASCII a-z and A-Z inside of the field name.
+
+### Sequential IDs
+
+When returning a resource, avoid using a sequential ID in the ID field. Using sequential IDs allows
+callers to potentially guess a particular resource which can lead to leaking information. We
+have a Polymorphic ID in our `infrahq/infra/api` package which you should use instead.
+
+### Payload Structure
+
+When you are returning a payload, keep the following things in mind:
+  * Keep things simple
+  * Avoid unnecessary deep nested structures inside of resource payloads
+  * If you nest one resource into another resource, make certain that it maintains the same structure as if the resource
+    were on its own
+
+## Timestamps
+
+For timestamp fields, we follow [RFC 3339](https://datatracker.ietf.org/doc/html/rfc3339) which overlaps
+with ISO 8601. Timestamps are always stored in in UTC form. The client is responsible for converting any
+timestamp from UTC to local time.
+
+Timestamp fields follow the form:
+
+```YYYY-MM-DDThh:mm:ss.SSSZ```
+
+
+Date Format:
+  * `[YYYY]` represents the four digit year
+  * `[MM]` represents to two digit month (pad with zeros)
+  * `[DD]` is the two digit day of the month (pad with zeros)
+
+Time Format:
+  * `T` represents the beginning of the time part of the timestamp (as opposed to the date)
+  * `[hh]` is the hour of the day (00 through 23)
+  * `[mm]` is the minute of the hour (00 through 59)
+  * `[ss]` represents the number of seconds (00 through 59)
+  * `[SSS]` represents the number of milliseconds (000 through 999)
+  * `Z` denotes that the timestamp is in UTC (zulu time)
+
+## Errors and Status Codes
+
+Always use the appropriate status code and write good error and status messages when
+returning a response. Writing good error messages helps the user determine when something
+went wrong, and it helps clients determine how to recover properly from an error.
+
+Error messages should never include details about how the internals of the server work;
+these tend to be non-sensical and don't help the client recover successfuly from an
+issue.
+
+We currently use the following status codes when the API returns:
+
+| Status | Type                  |
+| :----- | :---                  |
+| 200    | Success               | 
+| 201    | Created               |
+| 204    | No content            |
+| 400    | Bad request           |
+| 401    | Unauthorized          |
+| 403    | Forbidden             |
+| 404    | Not found             |
+| 409    | Duplicate             |
+| 500    | Internal server error |
+| 502    | Bad gateway           |
+
+
+Currently unused status codes include:
+
+| Status | Type                  |
```

Comment:
These unused status codes feel like they will change, and we probably won't realize to update this document.